sdk/python/kfp/v2/compiler_cli_tests/test_data/pipeline_with_various_io_types.json

```diff
@@ -1,118 +1,218 @@
+                  "taskOutputParameter": {
+                    "producerTask": "task-upstream",
+                    "outputParameterKey": "output_1"
+                  }
+                }
+              },
+              "artifacts": {
+                "input_c": {
+                  "taskOutputArtifact": {
+                    "outputArtifactKey": "output_3",
+                    "producerTask": "task-upstream"
+                  }
+                },
+                "input_e": {
+                  "taskOutputArtifact": {
+                    "outputArtifactKey": "output_5",
+                    "producerTask": "task-upstream"
+                  }
+                },
+                "input_b": {
+                  "taskOutputArtifact": {
+                    "outputArtifactKey": "output_2",
+                    "producerTask": "task-upstream"
+                  }
+                },
+                "input_g": {
+                  "taskOutputArtifact": {
+                    "producerTask": "task-upstream",
+                    "outputArtifactKey": "output_7"
+                  }
+                },
+                "input_d": {
+                  "taskOutputArtifact": {
+                    "producerTask": "task-upstream",
+                    "outputArtifactKey": "output_4"
+                  }
+                },
+                "input_f": {
+                  "taskOutputArtifact": {
+                    "producerTask": "task-upstream",
+                    "outputArtifactKey": "output_6"
+                  }
+                }
+              }
+            },
+            "componentRef": {
+              "name": "comp-downstream"
             }
           },
+          "task-upstream": {
+            "componentRef": {
+              "name": "comp-upstream"
+            },
+            "taskInfo": {
+              "name": "task-upstream"
+            },
+            "inputs": {
+              "parameters": {
+                "input_1": {
+                  "componentInputParameter": "input1"
+                },
+                "input_4": {
+                  "componentInputParameter": "input4"
+                },
+                "input_3": {
+                  "componentInputParameter": "input3"
+                },
+                "input_2": {
+                  "runtimeValue": {
+                    "constantValue": {
+                      "doubleValue": 3.1415926
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      },
+      "inputDefinitions": {
+        "parameters": {
+          "input4": {
+            "type": "STRING"
+          },
+          "input1": {
+            "type": "STRING"
+          },
+          "input3": {
+            "type": "STRING"
+          }
+        }
+      }
+    },
+    "components": {
+      "comp-downstream": {
+        "inputDefinitions": {
           "artifacts": {
-            "output_3": {
+            "input_g": {
```

Comment:
Nit: In the future you might look into stabilizing the output ordering. I'm not yet sure where the randomization occurs since in Python 3.6+ the dictionaries preserver the order.